src/framework/components/render/component.js

```diff
@@ -852,6 +868,10 @@ class RenderComponent extends Component {
         this._evtLayersChanged?.off();
         this._evtLayersChanged = null;
 
+        if (this._rootBone) {
+            this._clearSkinInstances();
```

Comment:
Ideally this would not be here, but I see how it makes it easier / more consistent, and the impact is low.